### PR TITLE
[client] Add support for `ray.timeline()`

### LINKED ISSUE #15192: [ray client] Support public API: ray.timeline()

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
* Add `ray.timeline()` support to Ray Client

## Related issue number

Closes #15192

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by DmitriGekhtman — APPROVED** [review]

File: python/ray/util/client/api.py [hunk 1]
```diff
@@ -247,6 +253,18 @@ def get_runtime_context(self):
     def get_gpu_ids(self) -> list:
         return []
 
+    def timeline(self, filename: Optional[str] = None) -> Optional[List[Any]]:
+        logger.info("Timeline will include events from other clients using "
```
Comment (AmeerHajAli) [inline python/ray/util/client/api.py:257]: is this more like a warning?

File: python/ray/util/client/api.py [hunk 2]
```diff
@@ -247,6 +253,18 @@ def get_runtime_context(self):
...
+                    "this server.")
+        import ray.core.generated.ray_client_pb2 as ray_client_pb2
```
Comment (AmeerHajAli) [inline python/ray/util/client/api.py:259]: circular dependency?

**Review by AmeerHajAli — APPROVED** [review]
LGTM.

File: python/ray/util/client/api.py (hunk 1 above)
Comment (ijrsvt) [reply python/ray/util/client/api.py:257]: Sure!

File: python/ray/util/client/api.py (hunk 2 above)
Comment (ijrsvt) [reply python/ray/util/client/api.py:259]: Yeah, let me add the comment that is used elsewhere in this file.

**Comment by ijrsvt** [comment]
Docker Build failed on BuildKite (a known issue)
Mac Wheels failed (a known issue as well recently).

**Pull request merged**